contracts/TokenFactory.sol

```diff
@@ -199,13 +205,8 @@ contract TokenFactory is ReentrancyGuard, Ownable {
 
         collateral[tokenAddress] = tokenCollateral;
 
-        // store collateralByDay for winner detecting
-        uint256 currentDay = startOfCurrentDay();
-        uint256 tokenCollateralToday = collateralByDay[currentDay][
-            tokenAddress
-        ];
-        tokenCollateralToday += contributionWithoutFee;
-        collateralByDay[currentDay][tokenAddress] = tokenCollateralToday;
+        // store collateralById for winner detecting
+        collateralById[currentCompetitionId][tokenAddress] += contributionWithoutFee;
```

Comment:
Note: when this function is called by `burnTokenAndMintWinner`, `currentCompetitionId` will represent a different competition id than the id which the token was initially created (and determined as winner)

Reply:
It is probably fine, just something should be paying attention to, maybe add in comment